python/lsst/analysis/tools/actions/scalar/scalarActions.py

```diff
@@ -432,5 +435,13 @@ def __call__(self, data: KeyedData, **kwargs) -> Scalar:
         scalarA = self.actionA(data, **kwargs)
         scalarB = self.actionB(data, **kwargs)
         if scalarB == 0:
-            raise ValueError("Denominator is zero!")
-        return scalarA / scalarB
+            if scalarA == 0:
+                log.warning("Both numerator and denominator are zero! Returning NaN.")
+                return np.nan
+            else:
+                log.warning(
+                    "Non-zero scalar divided by zero! Returning %sInf." % ("+" if scalarA > 0 else "-")
```

Comment:
```suggestion
                    "Non-zero scalar divided by zero! Returning %sInf.", "+" if scalarA > 0 else "-"
```
Do not use string formatting in a log message. Logs use deferred formatting.

Additionally, it looks like you are calculating the +/- twice. Once here in the log message and once below in the np.sign() so it might be more efficient to calculate the return value up front and then use that return value directly in the log message.

Something like:
```python
value = np.sign(scalarA) * np.inf
log.warning("Non-zero.... Returning %f", value
return value
```